src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/ReferenceRegistries/SecuritySchemeReferenceRegistry.cs

```diff
@@ -0,0 +1,356 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Exceptions;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Extensions;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Models.KnownStrings;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.ReferenceRegistries
+{
+    /// <summary>
+    /// Reference Registry for <see cref="OpenApiSecurityScheme"/>.
+    /// </summary>
+    public class SecuritySchemeReferenceRegistry : ReferenceRegistry<XElement, OpenApiSecurityScheme>
+    {
+        /// <summary>
+        /// The dictionary containing all references of the given security scheme.
+        /// </summary>
+        public override IDictionary<string, OpenApiSecurityScheme> References { get; } =
+            new Dictionary<string, OpenApiSecurityScheme>();
+
+        internal List<string> Scopes { get; private set; }
```

Comment:
>List [](start = 17, length = 4)

nit, IList

Reply:
doing AddRange below so that is why list

---
In reply to: [221026480](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/157#discussion_r221026480) [](ancestors = 221026480)